### ISSUE TITLE
Add Runge-Kutta benchmarks

### ISSUE THREAD

**Issue by certik** [fortran-lang/benchmarks#5]
More details with Fortran and Julia code available at:

https://fortran-lang.discourse.group/t/improving-fortran-results-in-the-julia-micro-benchmarks/198/17

**Comment by ivan-pi** [comment]
I found a few more ODE integration benchmarks of Fortran vs Julia: https://github.com/PIK-ICoN/NetworkDynamicsBenchmarks

**Comment by awvwgk** [comment]
While I'm happy to see fpm used outside the Fortran community, it also shows some risks when not used properly.
In this case, it looks like they used Haskell fpm's flag feature to link against pthread:
```
fpm build --flag -ldl --flag -lpthread
```
Luckily for them, this also disables the debugging flags and runtime checks from the debug profile.
Still, the Fortran timings could improve by actually enabling optimizations for the benchmark run.

**Comment by ivan-pi** [comment]
Good catch! I didn't even pay attention to the fact they are using `fpm`.

**Comment by certik** [comment]
Shows that we need to improve fpm to be usable easily by the outside community.

**Comment by Lima-Lima** [comment]
Hi all! I put that Fortran benchmark together. It was my first foray into FPM. I had expected it to go more smoothly. 

I couldn't determine how to choose flags using the Fortran FPM, thus I used the Haskell version. Optimizations flags were also added using the --flags feature (I explicitly indicated all my desired flags). 

Can you clarify: Does using a '--flag' disable all existing flags in the --debug profile? Is this documented somewhere?

Thanks for any input or improvements regarding use of the fpm tool. I'm pleased to see effort into a unified Fortran package manager and had hopes it would go a bit more smoothly than it did. It seems to me most features are undocumented. You'll see I also hacked the Flibs sqlite wrapper to use FPM for this little project, but it isn't using the fpm build system in a good way: https://github.com/Lima-Lima/fsqlite

**Comment by arjenmarkus** [comment]
Hi Lucas,

as the author of fsqlite (and Flibs) I am interested in the issues you
found. I have not tried it myself yet, but it should be a nice exercise -
for both fsqlite and fpm :).

Op di 30 mrt. 2021 om 08:41 schreef Lucas ***@***.***>:
